src/scss/components/switch/_index.scss

```diff
@@ -0,0 +1,165 @@
+@import "./config";
+
+hx-switch-control {
+  -ms-grid-column-align: start;
+  -ms-grid-columns: auto 1fr;
+  -ms-grid-rows: auto auto;
+  display: -ms-grid;
+  display: grid;
+  grid-template-areas:
+    "ctrl lbl"
+    ".    help";
+  grid-template-columns: auto 1fr;
+  grid-template-rows: auto auto;
+  justify-items: start;
+  grid-area: facade;
+  margin-top: 0.25rem; // FIXME: fix with proper inline styling
+  position: relative;
+  width: 4.25rem;
+  height: 2rem;
+
+  > input {
+    opacity: 0;
+    width: 0;
+    height: 0;
+  }
+
+  > label > hx-switch {
+    position: absolute;
+    cursor: pointer;
+    top: 0;
+    left: 0;
+    right: 0;
+    bottom: 0;
+    -webkit-transition: 0.4s;
+    transition: 0.4s;
+    border-radius: 2.125rem;
+  }
+
+  > label > hx-switch::before {
+    position: absolute;
+    content: "";
+    height: 1.75rem;
+    width: 1.625rem;
+    -webkit-transition: 0.4s;
+    transition: 0.4s;
+    border-radius: 50%;
+  }
+}
+
+hx-switch-control > input[type="checkbox"] ~ label {
+  > hx-switch {
+    background-color: $gray-0;
+    border: 0.125rem solid $cyan-900;
+  }
+
+  > hx-switch[offlabel]::after {
+    content: attr(offlabel);
+    text-transform: uppercase;
+    margin-left: 1.688rem;
+    color: $cyan-900;
+    line-height: 1.75rem;
+    max-width: 1.813rem;
+    display: inline-block;
+    overflow: hidden;
+    white-space: nowrap;
+  }
+
+  > hx-switch::before {
+    background-color: $gray-0;
+    border: 0.125rem solid $cyan-900;
+    left: 0;
+  }
+}
+
+hx-switch-control > input[type="checkbox"]:checked ~ label {
+  > hx-switch {
+    background-color: $cyan-900;
+    @include hxSwitchControl($pseudo-state: focus);
+  }
+
+  > hx-switch[onlabel]::after {
+    content: attr(onlabel);
+    text-transform: uppercase;
+    margin-left: 0.625rem;
+    color: $gray-0;
+    max-width: 2.125rem;
+    display: inline-block;
+    overflow: hidden;
+    white-space: nowrap;
+  }
+
+  > hx-switch::before {
+    margin-left: 2.438rem;
+    height: 1.813rem;
+  }
+}
+
+.switch {
+  position: relative;
+  display: inline-block;
+  width: 4.25rem;
+  height: 2rem;
+}
+
+// ========== DISABLED ==========
+hx-switch-control > input[type="checkbox"]:disabled {
+  &:checked {
+    ~ label > hx-switch {
+      @include hxSwitchControl(disabled);
+    }
+  }
+
+  ~ label > hx-switch {
+    @include hxSwitchControl(disabled, unchecked);
+  }
+}
+
+// ========== INVALID ==========
+hx-switch-control > input[inValid] {
+  &:checked {
+    ~ label > hx-switch {
+      @include hxSwitchControl(error, null, checked);
+      @include hxSwitchControl(error, focus);
+    }
+  }
+
+  ~ label > hx-switch {
+    @include hxSwitchControl(error, null, unchecked);
+    @include hxSwitchControl(error, focus);
+  }
+}
+
+// --- start of hover css for hx-switch
+hx-switch-control {
+  #errorChkDemo:hover {
```

Comment:
This should be handled by the Switch component styles.

Try moving this to the `INVALID` block: 

```scss
// ========== INVALID ==========
hx-switch-control[error] {
 
  ....

  &:hover {
```